### ISSUE TITLE
esp32-S3 support? (US-99)

### ISSUE THREAD

**Issue by dvl36** [espressif/esp-privilege-separation#3]
Are there plans for esp32-**s**3 support?

**Comment by sachin0x18** [comment]
Hi @dvl36,

The support for ESP32-S3 is in pipeline. I will update here once the support for ESP32-S3 is merged into master branch.

**Comment by dvl36** [comment]
Hi @sachin0x18,

That's great!
What is the approximate time frame?

**Comment by sachin0x18** [comment]
@dvl36 We have merged preliminary support for ESP32-S3.

You can now try it out by commenting the following lines from example's sdkconfig.defaults

```
CONFIG_IDF_TARGET_ARCH_RISCV=y
CONFIG_IDF_TARGET="esp32c3"
CONFIG_IDF_TARGET_ESP32C3=y
```
 
After this, you need to run `idf.py set-target esp32s3` and then the examples should work on ESP32S3

Feel free to report any issues or difficulties

**Comment by chegewara** [comment]
Hi team,
it seems to work on S3, at least with blink example. 
I have 2 notes, not really complaining:
- from monitor console CTRL + T with CTRL + A flash only privilege app, not flashing user app
- here is a bit misleading log message:
```
Warning: 1/2 app partitions are too small for binary blink.bin size 0xce630:
  - Part 'user_app' 0/64 @ 0x110000 size 0x80000 (overflow 0x4e630)
[2/5] Performing build step for 'user_app'
```
Its with `partitions.csv` from example, where the partition is `user_app` and info is about `main_app` not fit.

Thanks

PS one more thing i forgot about; can we update example
https://github.com/espressif/esp-privilege-separation/blob/master/examples/blink/user_app/main/user_code.c#L28-L35
with this change, to use IO0 on S3:
```
#if CONFIG_IDF_TARGET_ESP32C3
#define WS2812_GPIO     8
#define BUTTON_IO       9
#elif CONFIG_IDF_TARGET_ESP32S3
#define WS2812_GPIO    48
#define BUTTON_IO       0
#endif
#define INTR_LED        2
#define BLINK_GPIO      10
```

**Comment by chegewara** [comment]
Hmm, 
is it required to use S3 in unicore mode?
```
CONFIG_FREERTOS_UNICORE=y
```
Because when i remove it from defconfig then app crashes.

Thanks

**Comment by sachin0x18** [comment]
Hi @chegewara,

> from monitor console CTRL + T with CTRL + A flash only privilege app, not flashing user app

Thanks for catching this. We will take a look into this.

> Warning: 1/2 app partitions are too small for binary blink.bin size 0xce630:
> - Part 'user_app' 0/64 @ 0x110000 size 0x80000 (overflow 0x4e630)
> [2/5] Performing build step for 'user_app'

The check_sizes.py script simply checks all the partitons with type `app` and that is the reason why it prints this warning. It doesn't affect the functioning but yes it would be good to handle this in the future.

> PS one more thing i forgot about; can we update example

Oh yes, I missed changing the BOOT button. I will fix this. Thanks for catching it

> is it required to use S3 in unicore mode?

Yes. For now, privilege separation on S3 will only work on unicore

**Comment by NB-player** [comment]
Hi team , 
I am a beginner.
 I followed the tutorial in the example and the official documentation to compile my sample program for my esp32s3, but I still encountered the following error.

Could you please tell me if this error is due to an incorrect installation process for the patch? If so, how can I successfully compile and burn a sample program using this project? If not, what should I do to solve this problem?

Here is the error code:
`-- Project is not inside a git repository, or git repository has no commits; will not use 'git describe' to determine PROJECT_VER.
-- Building ESP-IDF components for target esp32s3
CMake Error at /usr/esp/esp-idf/tools/cmake/build.cmake:201 (message):
  Failed to resolve component 'esp_priv_access'.
Call Stack (most recent call first):
  /usr/esp/esp-idf/tools/cmake/build.cmake:227 (__build_resolve_and_add_req)
  /usr/esp/esp-idf/tools/cmake/build.cmake:507 (__build_expand_requirements)
  /usr/esp/esp-idf/tools/cmake/project.cmake:384 (idf_build_process)
  CMakeLists.txt:14 (project)`

**Comment by sachin0x18** [comment]
Hi @NB-player

Can you please elaborate a bit more on which example are you trying to build ? Also, it would be better if you can create a new issue and mention all the steps that you followed in it

**Comment by mahavirj** [comment]
Closing as the original question regarding S3 support has been answered here.
